scalastyle-config.xml

```diff
@@ -150,6 +150,19 @@ This file is divided into 3 sections:
       // scalastyle:on println]]></customMessage>
   </check>
 
+  <check customId="hadoopconfiguration" level="error" class="org.scalastyle.file.RegexChecker" enabled="true">
+    <parameters><parameter name="regex">spark(.sqlContext)?.sparkContext.hadoopConfiguration</parameter></parameters>
```

Comment:
can we just match on hadoopConfiguration ?

Reply:
I think that would be tricky:
1. In core module the function `spark.sessionState.newHadoopConf()` can't be used.
2. Run `find . -name *.scala|xargs grep hadoopConfiguration |wc -l` and there are 100 appearances.

Reply:
Even with such loose policy, we have to wrap code block with "// scalastyle:off hadoopconfiguration" in some test cases for setting the hadoop conf.